src/main/java/com/aws/greengrass/lifecyclemanager/GenericExternalService.java

```diff
@@ -335,6 +352,9 @@ protected synchronized void startup() throws InterruptedException {
                 }
             }
         }, lifecycleProcesses);
+        if (result.getRight() != null) {
+            systemResourceController.addComponentProcess(this, result.getRight().getProcess());
+        }
```

Comment:
I think it will be better to do this after the Errored / NothingDone checks below on line 359-364, otherwise it's unnecessarily initializing cgroups in those cases

Reply:
The null check should have prevented that.